### PR TITLE
add information and instructions concerning LXD storage

### PR DESCRIPTION
**Pull request by ru-fu**
Signed-off-by: Ruth Fuchss <ruth.fuchss@canonical.com>

### CODE REVIEW

File: exp/anbox-cloud.md [hunk 1]
```diff
@@ -77,6 +80,42 @@ The main components powering the streaming stack in Anbox Cloud are:
...
+| ams-cdr3292j841mb72gksog | RUNNING | 192.168.96.1 (eth0)    |      | CONTAINER | 0         | lxd0     |
+|                          |         | 192.168.250.1 (anbox0) |      |           |           |          |
++--------------------------+---------+------------------------+------+-----------+-----------+----------+
+| juju-73e292-0            | RUNNING | 240.0.233.254 (eth0)   |      | CONTAINER | 0         | lxd0     |
++--------------------------+---------+------------------------+------+-----------+-----------+----------+
+| juju-73e292-1            | RUNNING | 240.0.98.38 (eth0)     |      | CONTAINER | 0         | lxd0     |
++--------------------------+---------+------------------------+------+-----------+-----------+----------+
+| juju-73e292-2            | RUNNING | 240.0.61.208 (eth0)    |      | CONTAINER | 0         | lxd0     |
++--------------------------+---------+------------------------+------+-----------+-----------+----------+
+| juju-34631c-0            | RUNNING | 240.0.180.30 (eth0)    |      | CONTAINER | 0         | lxd0     |
```
Comment (ru-fu) [inline exp/anbox-cloud.md:104]: To we need to change/hide the IP addresses here?

File: howto/install/deploy-bare-metal.md [hunk 2]
```diff
@@ -74,6 +74,37 @@ The `anbox-cloud` bundle requires two additional machines to host the load balan
...
+There are different ways of configuring a dedicated block storage device:
+
+- Use a storage device defined by Juju (see the *Customise storage* section in [How to deploy Anbox Cloud with Juju](https://discourse.ubuntu.com/t/install-with-juju/17744#customise-storage) for instructions)
+- Use a dedicated storage device (see [Dedicated storage device](#dedicated-storage-device) below)
+- Use an existing LXD storage pool (see [Existing storage pool](#existing-storage-pool) below)
+
+<a name="dedicated-storage-device"></a>
+### Dedicated storage device
+
+To use a dedicated storage device that is not defined by Juju for LXD storage, set the [`storage_device`](https://charmhub.io/ams/configure#storage_device) configuration on the AMS charm to the path of the storage device.
```
Comment (ru-fu) [inline howto/install/deploy-bare-metal.md:91]: Is this the correct way of doing it?

File: howto/install/deploy-bare-metal.md [hunk 3]
```diff
@@ -74,6 +74,37 @@ The `anbox-cloud` bundle requires two additional machines to host the load balan
...
+[note type="information" status="Important"]
+The path to the dedicated storage device must be identical for all machines that are part of the cluster.
+[/note]
+
+You do not need to prepare the storage device in any way. AMS takes care of creating the LXD storage pool on the device.
+
+<a name="existing-storage-pool"></a>
+### Existing storage pool
+
+To use an existing LXD storage pool, set the [`storage_pool`](https://charmhub.io/ams/configure#storage_pool) configuration on the AMS charm to the name of the LXD storage pool that you want Anbox Cloud to use.
```
Comment (ru-fu) [inline howto/install/deploy-bare-metal.md:102]: Is this the correct way of doing it?

File: howto/manage/resize-storage.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+However, if you run into a situation where you need to grow the LXD storage pool, it depends on your setup if this is actually possible.
+
+- **Loop-backed storage pool**
+
+  If you use a loop file for your LXD storage, you can increase its size by following the instructions for ZFS in [Resize a storage pool](https://linuxcontainers.org/lxd/docs/latest/howto/storage_pools/#resize-a-storage-pool) in the LXD documentation.
+
+  After resizing the storage pool, you must restart AMS.
+- **Dedicated block device**
+
+  You cannot resize a dedicated block device. In this case, you must replace the block device with a bigger one. Make sure to mount the new block device at the same location as the old one.
```
Comment (ru-fu) [inline howto/manage/resize-storage.md:14]: Does this make sense?

File: howto/manage/resize-storage.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+- **EBS volume**
+
+  While it is generally possible to resize an EBS volume, it is usually easier to replace the existing volume with a new one that has a different size. Make sure to mount the new EBS volume at the same location as the old one.
```
Comment (ru-fu) [inline howto/manage/resize-storage.md:17]: Does this make sense (especially the part about mounting)?

File: exp/anbox-cloud.md (hunk 1 above)
Comment (morphis) [reply exp/anbox-cloud.md:104]: No, these are fine to keep

File: howto/install/deploy-juju.md [hunk 6]
```diff
@@ -156,6 +162,35 @@ To deploy, add `--overlay overlay.yaml` to your deploy command. For example:
...
+
+   The `ams-lxd` charm defines an optional storage pool (see the `metadata.yaml` file in the `ams-lxd` charm). To make the LXD charm use this storage pool, you must configure a [storage constraint](https://juju.is/docs/olm/storage-constraint) for it.
+
+   For example, to use the AWS `ebs-ssd` Juju storage pool for LXD storage, use an overlay file with the following storage constraint:
+
+   ```
+   applications:
+     lxd:
+       storage:
+         pool: ebs-ssd,100GB,1
```
Comment (morphis) [inline howto/install/deploy-juju.md:189]: Do we need to point out that adding more than one volume will cause an error?

**Review by morphis — CHANGES_REQUESTED** [review]

File: howto/install/deploy-bare-metal.md [hunk 7]
```diff
@@ -74,6 +74,37 @@ The `anbox-cloud` bundle requires two additional machines to host the load balan
...
+To use a dedicated storage device that is not defined by Juju for LXD storage, set the [`storage_device`](https://charmhub.io/ams/configure#storage_device) configuration on the AMS charm to the path of the storage device.
+
+[note type="information" status="Important"]
+The path to the dedicated storage device must be identical for all machines that are part of the cluster.
+[/note]
+
+You do not need to prepare the storage device in any way. AMS takes care of creating the LXD storage pool on the device.
+
+<a name="existing-storage-pool"></a>
+### Existing storage pool
```
[the remark this answers is not in the document]
Comment (ru-fu) [reply howto/install/deploy-bare-metal.md:100]: I changed the order now to have the existing storage pool above the dedicated block device.
But do we really want to recommend this alternative? Or do we rather recommend using Juju-defined storage?

File: howto/install/deploy-juju.md [hunk 8]
```diff
@@ -156,6 +162,35 @@ To deploy, add `--overlay overlay.yaml` to your deploy command. For example:
...
+The easiest way to do this is to use a storage device defined by Juju:
+
+1. Decide which Juju storage pool you want to use.
+
+   See [View the available storage pools](https://juju.is/docs/olm/manage-storage-pools#heading--view-the-available-storage-pools) in the Juju documentation for instructions on how to display existing storage pools. If you are running on AWS, for example, you can use the existing `ebs-ssd` pool.
+
+   If you want to use a new pool, see [Create a storage pool](https://juju.is/docs/olm/manage-storage-pools#heading--create-a-storage-pool) in the Juju documentation for instructions. It can also be useful to create a Juju storage pool if you want to use a specific volume type that is optimised for your setup. For example, AWS provides [EBS volume types](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html) that are optimised for different purposes.
+
+   If you decide to create a pool, make sure to do so before you start the deployment.
+1. Configure the `ams-lxd` charm to use the Juju storage pool.
```
[the remark this answers is not in the document]
Comment (ru-fu) [reply howto/install/deploy-juju.md:179]: No, Markdown allows using "1." for all steps - that way, you don't need to worry about numbering when adding or moving a step.

File: howto/manage/resize-storage.md (hunk 4 above)
Comment (ru-fu) [reply howto/manage/resize-storage.md:14]: Added the information that it means re-deploying the node.

File: howto/install/deploy-juju.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (morphis) [reply howto/install/deploy-juju.md:179]: Ok

File: howto/install/deploy-bare-metal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (morphis) [reply howto/install/deploy-bare-metal.md]: For bare metal using juju defined storage isn't a great option as Juju does not provide good support for it without a cloud provider (which we don't have for bare metal)

**Review by morphis — APPROVED** [review]
Thanks, this is really looking good!

File: howto/install/deploy-bare-metal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ru-fu) [reply howto/install/deploy-bare-metal.md]: OK, moved that option to the bottom then and added a "recommended" for the storage pool. :)

**Pull request merged**